pkg/collector/corechecks/system/gpu/nvidia/jetson/jetson.go

```diff
@@ -119,56 +120,27 @@ func (c *JetsonCheck) processTegraStatsOutput(tegraStatsOuptut string) error {
 
 // Run executes the check
 func (c *JetsonCheck) Run() error {
-	// Kill tegrastats if it runs for twice as long as the interval we specified, to avoid blocking
-	// the check forever
-	ctx, cancel := context.WithTimeout(context.Background(), 2*tegraStatsInterval*time.Millisecond)
-	defer cancel()
-	cmd := exec.CommandContext(ctx, c.tegraStatsPath, c.commandOpts...)
-
-	// Parse the standard output for the stats
-	stdout, err := cmd.StdoutPipe()
-	if err != nil {
-		return err
+	tegraStatsCmd := fmt.Sprintf("%s %s", c.tegraStatsPath, strings.Join(c.commandOpts, " "))
+	cmdStr := fmt.Sprintf("(%s) & pid=$!; (sleep %d && kill -9 $pid)", tegraStatsCmd, int((2*tegraStatsInterval).Seconds()))
+	var cmd *exec.Cmd
+	if c.useSudo {
+		// -n, non-interactive mode, no prompts are used
+		cmd = exec.Command("sudo", "-n", "sh", "-c", cmdStr)
+	} else {
+		cmd = exec.Command("sh", "-c", cmdStr)
 	}
-	go func() {
-		in := bufio.NewScanner(stdout)
-		if in.Scan() {
-			// We only need to read one line
-			line := in.Text()
-			log.Debugf("tegrastats: %s", line)
-			if err = c.processTegraStatsOutput(line); err != nil {
-				log.Error(err)
-			}
-		} else {
-			log.Warnf("tegrastats did not produce any output")
-		}
-		// Tegrastats keeps running forever, so kill it after trying to read
-		// one line of output
-		err = cmd.Process.Signal(os.Kill)
-		if err != nil {
-			log.Errorf("unable to stop %s check: %s", checkName, err)
-		}
-	}()
 
-	// forward the standard error to the Agent logger
-	stderr, err := cmd.StderrPipe()
-	if err != nil {
-		return err
-	}
-	go func() {
-		in := bufio.NewScanner(stderr)
-		for in.Scan() {
-			log.Error(in.Text())
+	// Since we kill the process, err is always going to be != nil
+	tegrastatsOutput, _ := cmd.Output()
```

Comment:
nit: We could compare the error to [`exec.ErrNotFound`](https://golang.org/pkg/os/exec/#pkg-variables) to at least report the error in that case even if we can't report it everytime.

Reply:
Yeah, probably it will only report with `sudo`/`sh`/`kill` missing but that way we are not completely blind in that case (`sudo` at least could be missing in some containerized environments I think)